src/components/MaterialUI/hocs/withDeferredPopper.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from "react";
+import Popper from '@material-ui/core/Popper';
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
+import {
+  isString,
+  isObject,
+  isStringNullOrWhitespace,
+  isReactComponent,
+} from "../../../utils/propertyValidator";
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(theme => ({
+  popper: {
+    zIndex: 1,
+    '&[x-placement*="bottom"] $arrow': {
+      top: 0,
+      left: "50%",
+      transform: "translateX(-50%) rotate(45deg)",
+      marginTop: '-0.25em',
+      width: '0.4em',
+      height: '0.4em',
+      borderColor: `${theme.palette.grey.icon} transparent transparent ${theme.palette.grey.icon}`,
+    },
+    '&[x-placement*="top"] $arrow': {
+      bottom: 0,
+      left: "50%",
+      transform: "translateX(-50%) rotate(45deg)",
+      marginBottom: '-0.25em',
+      width: '0.4em',
+      height: '0.4em',
+      borderColor: `transparent ${theme.palette.grey.icon} ${theme.palette.grey.icon} transparent`,
+    },
+    '&[x-placement*="right"] $arrow': {
+      left: 0,
+      top: "50%",
+      marginLeft: '-0.25em',
+      transform: "translateY(-50%) rotate(45deg)",
+      width: '0.4em',
+      height: '0.4em',
+      borderColor: `transparent transparent ${theme.palette.grey.icon} ${theme.palette.grey.icon}`,
+    },
+    '&[x-placement*="left"] $arrow': {
+      right: 0,
+      top: "50%",
+      marginRight: '-0.25em',
+      transform: "translateY(-50%) rotate(45deg)",
+      width: '0.4em',
+      height: '0.4em',
+      borderColor: `${theme.palette.grey.icon} ${theme.palette.grey.icon} transparent transparent`,
+    },
+  },
+  arrow: {
+    position: "absolute",
+    border: "1px solid",
+    backgroundColor: theme.palette.background.paper,
+  }
+}));
+
+export const Arrow = () => {
+  const classes = useStyles();
+
+  return <div className={classes.arrow} />
+}
+
+const withDeferredPopper = Comp => ({ popperValue, ...props }) => {
+  const classes = useStyles();
+
+  const [popperState, setPopperState] = useState({
+    isDisplayed: false,
+    anchorElement: null
+  });
+
+  const defaultComponent = (
+    <Comp onClick={(event) => togglePopper(event)} {...props} />
+  );
+
+  const togglePopper = function (event) {
+    const isDisplayed = !popperState.isDisplayed;
+    const anchorElement = event.currentTarget;
+    setPopperState({
+      isDisplayed: isDisplayed,
+      anchorElement: anchorElement
+    })
+  };
+
+  const clickAwayHandler = function () {
+    setPopperState({
+      isDisplayed: false,
+      anchorElement: null
+    })
+  }
+
+  if (popperValue == null) return <Comp {...props} />;
```

Comment:
In fact before mounting (what happens after return) it doesn't hurt performance to execute all those lines before. From the code readability probably it would be better to handle that as early as possible.

Reply:
Oh, I was not saying it would hurt performance, it was only about sequence of operations as I personal preference of mine.